packages/sonarwhal/src/lib/types/resourceerror.ts

```diff
@@ -0,0 +1,13 @@
+import { ResourceErrorStatus } from '../enums/errorstatus';
+
+export class ResourceError extends Error {
+    public status: ResourceErrorStatus;
+
+    public constructor(error: Error, status: ResourceErrorStatus, message?: string) {
+        super(error.message);
```

Comment:
This signature looks a bit weird, especially when you do later a:

```js
throw new ResourceError(new Error(`Resource ${name} isn't compatible with current sonarwhal version`), ResourceErrorStatus.NotCompatible);
```

We shouldn't have to create a new error outside the constructor. It looks like there are a few things going on with TS and inheriting from built-ins in TS:

* https://www.metachris.com/2017/01/custom-errors-in-typescript-2.1/
* https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget


But those seem to be from TS 2.1 and 2.2, not sure a bout the current status for us in node. In any case, I think we should have something like:

```ts
public constructor(error: Error | string, status: ResourceErrorStatus) { }
```

or even 

```ts
public constructor(error: string, status: ResourceErrorStatus) { }
```

to simplify things.